nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/provider/flow/git/GitFlowMetaData.java

```diff
@@ -136,36 +136,50 @@ private Repository openRepository(final File gitProjectRootDir) throws IOExcepti
     }
 
     private static boolean hasAtLeastOneReference(Repository repo) {
-        logger.info(repo.toString());
+        logger.info("Checking references for repository {}", repo.toString());
         for (Ref ref : repo.getAllRefs().values()) {
-            if (ref.getObjectId() == null)
+            if (ref.getObjectId() == null) {
                 continue;
+            }
             return true;
         }
         return false;
     }
 
+    /**
+     * Check if provided local repository exists or not at given 'Flow Storage Directory' in providers.xml
+     * @param localRepo file object of 'Flow Storage Directory'
+     * @return true if local repository exists else false
+     * @throws IOException if .git of local repository cannot be opened
+     */
     public boolean localRepoExists(File localRepo) throws IOException {
         if (!localRepo.isDirectory()) {
             logger.info("{} is not a directory or does not exist.", localRepo.getPath());
             return false;
         }
 
         if (RepositoryCache.FileKey.isGitRepository(new File(localRepo.getPath()+"/.git"), FS.DETECTED)) {
-            Git git = Git.open(new File(localRepo.getPath() + "/.git"));
-            Repository repository = git.getRepository();
+            final Git git = Git.open(new File(localRepo.getPath() + "/.git"));
+            final Repository repository = git.getRepository();
             logger.info("Checking for git references in {}", localRepo.getPath());
-            boolean referenceExists = hasAtLeastOneReference(repository);
-            if (referenceExists)
+            final boolean referenceExists = hasAtLeastOneReference(repository);
+            if (referenceExists) {
                 logger.info("{} local repository exists with references so no need to clone remote", localRepo.getPath());
+            }
             // Can be an empty repository if no references are present should we pull from remote?
             return true;
         }
         return false;
     }
 
+    /**
+     * Validate if provided 'Remote Clone Repository' in providers.xml exists or not. If remote repository
```

Comment:
Better:

Validate if the provided 'Remote Clone Repository' configuration in the providers.xml exists or not. If the remote repository does not exist, an {@link IllegalArgumentException} will be thrown.